intellij/src/saros/intellij/ui/tree/SessionAndContactsTreeView.java

```diff
@@ -105,6 +111,21 @@ public SessionAndContactsTreeView(@NotNull Project project) {
 
     connectionHandler.addConnectionStateListener(connectionStateListener);
 
+    contactsUpdateListener =
+        (contact, updateType) -> {
+          if (project.isDisposed()) {
```

Comment:
I do not know the threading requirements but such logic would not work in SWT. 

This is not called by the UI thread (which basically does a dispose) and so even if this returns false the dispose may happen just after the check. I guess Swing is more graceful in handling this but you would be surprised what can happen in SWT when using this pattern.

Reply:
Hmmmm. The only alternative I see would be to move the check into the `UIUtil.invokeLaterIfNeeded(...)` block in `renderConnected()` as I would assume that the project disposal is also run on the EDT. That would ensure that the action is aborted if the disposal is run first.

Also, this shouldn't really hurt either way. The logic called in the listener just updates the Saros view, which should not be dependent on any project specific content but rather should just deal with the Saros state. And, after the action took place, the objects should still be garbage-collectible either way.